iphone/Maps/UI/BottomMenu/Menu/BottomMenuPresenter.swift

```diff
@@ -44,53 +49,50 @@ extension BottomMenuPresenter {
   }
 
   func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-    section == Sections.layers.rawValue ? 1 : CellType.allCases.count - (disableDonate ? 1 : 0)
-  }
-
-  private func correctedRow(_ row: Int) -> Int {
-    disableDonate && row >= CellType.donate.rawValue ? row + 1 : row
+    switch sections[section] {
+    case .layers:
+      return 1
+    case .items:
+      return menuCells.count
+    }
   }
   
   func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-    if indexPath.section == Sections.layers.rawValue {
+    switch sections[indexPath.section] {
+    case .layers:
       let cell = tableView.dequeueReusableCell(cell: BottomMenuLayersCell.self)!
       cell.onClose = { [weak self] in self?.onClosePressed() }
       return cell
+    case .items:
+      let cell = tableView.dequeueReusableCell(cell: BottomMenuItemCell.self)!
+      switch menuCells[indexPath.row] {
+      case .addPlace:
+        let enabled = MWMNavigationDashboardManager.shared().state == .hidden && FrameworkHelper.canEditMapAtViewportCenter()
+        cell.configure(imageName: "ic_add_place",
+                       title: L("placepage_add_place_button"),
+                       enabled: enabled)
+      case .recordTrack:
+        trackRecorder.addObserver(cell, trackRecordingStateDidChange: { cell.setState($0) })
```

Comment:
Is observer here needed if menu will always be hidden after pressing Start or Stop recording?

Reply:
The observer will be automatically removed from the observers list because it is stored as weak reference.

Reply:
Observing is removed because we hide the screen on start/stop/continue recording